docker/container_test.go

```diff
@@ -0,0 +1,97 @@
+package docker
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/suite"
+)
+
+type ContainerTestSuite struct {
+	suite.Suite
+	cli *client.Client
+}
+
+func (suite *ContainerTestSuite) SetupSuite() {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	suite.NoError(err)
+	suite.cli = cli
+}
+
+func (suite *ContainerTestSuite) TearDownSuite() {
+	suite.cli.Close()
+}
+
+func (suite *ContainerTestSuite) TearDownTest() {
+	ctx := context.Background()
+	volumes, err := suite.cli.VolumeList(ctx, volume.ListOptions{})
+	suite.NoError(err)
+
+	for _, v := range volumes.Volumes {
+		err = suite.cli.VolumeRemove(ctx, v.Name, true)
+		suite.NoError(err)
+	}
+
+	images, err := suite.cli.ImageList(ctx, image.ListOptions{})
+	suite.NoError(err)
+
+	for _, img := range images {
+		_, err = suite.cli.ImageRemove(ctx, img.ID, image.RemoveOptions{Force: true})
+		suite.NoError(err)
+	}
+}
+
+func (suite *ContainerTestSuite) TestContainerStartHandle() {
+	first_timestamp := time.Now()
+	dm := NewDockerMonitor(context.Background(), 0, time.Hour)
```

Comment:
In the other tests you explicitly test if the `dm` variable is set. Is it intentional that this is missing here? 

Followup question: would it make sense to refactor that into the test setup function?